README.md

```diff
@@ -114,6 +114,63 @@ An example cURL command to start the connector in distributed mode is:
     data: {"id":"1543960661:0:2","clientId":"kafka-connect-ably-example","connectionId":"SuJTceISnT","timestamp":1623496744538,"encoding":"base64", "channel":"kafka-connect-ably-example","data":"bWVzc2FnZSAz","name":"sink"}
     ```
 
+#### Publishing messages with schema
+
+Ably Kafka connector supports messages with schema by converting schematic messages to JSON. Produced JSON is then passed as data to Ably message. JSON is produced by mapping schema field names to their values.
+For example if you use [Kafka Connect Avro Converter](https://www.confluent.io/hub/confluentinc/kafka-connect-avro-converter) and register an Avro schema to your [schema registry](https://docs.confluent.io/platform/current/schema-registry/connect.html), you can use the following. 
+
+Based on [docker-compose](/docker-compose.yml),  produce a set of test messages specifying `value.schema` property to a valid [Avro Schema](https://avro.apache.org/docs/current/spec.html) and specifying a valid`schema.registry.url` using the `kafka-avro-console-producer` CLI tool. Note that `kafka-avro-console-producer` tool is a part of schema-registry. Replace `<kafka-topic-name>` with one of the topics set in your configuration file.
+```shell
+ docker-compose exec -T schema-registry kafka-avro-console-producer \
+   --topic topic1 \
+   --broker-list kafka:9092 \
+   --property key.schema='{"type":"string"}' \
+   --property parse.key=true \
+   --property key.separator=":" \
+   --property value.schema='{"type":"record","name":"myrecord","fields":[{"name":"count","type":"int"}]}' \
+   --property schema.registry.url=http://schema-registry:8081 <<EOF
+"key1":{"count":1}
+"key2":{"count":2}
+"key3":{"count":3}
+EOF
+```
+
+You should receive following JSON messages where you subscribed:
+
+```json
+{"count":1}
+```
+
+```json
+{"count":2}
+```
+
+```json
+{"count":3}
+```
+
+You will also receive an Avro-formatted key base64 encoded in the extras. For example:
```

Comment:
I think we should just have one example that shows the data and the keys in extras coming through, rather than one example with just data, and a second with the keys.